aries_cloudagent/protocols/connections/v1_0/manager.py

```diff
@@ -840,13 +868,13 @@ async def get_endpoints(self, conn_id: str) -> Tuple[str, str]:
             Their endpoint for this connection
 
         """
-        connection = await ConnRecord.retrieve_by_id(self._session, conn_id)
-
-        wallet = self._session.inject(BaseWallet)
-        my_did_info = await wallet.get_local_did(connection.my_did)
+        async with self.profile.session() as session:
+            connection = await ConnRecord.retrieve_by_id(session, conn_id)
+            wallet = session.inject(BaseWallet)
+            my_did_info = await wallet.get_local_did(connection.my_did)
```

Comment:
doesn0t this need to be declared 1 scope above?

Reply:
Do you mean my_did_info?

Python allows this

```
if True:
  y = 10

print(y) -> 10
```

So basically function is the scope in which all vars are declared

Reply:
Evil! :) Resolving.